### ISSUE TITLE
Request for an initial version of a Dortania OpenCore Guide for Alder Lake

### ISSUE THREAD

**Issue by chriswayg** [dortania/bugtracker#257]
* Guide in question: Dortania's OpenCore Install Guide

* Link to page with the issue:
Comet Lake is the last Intel guide. Alder Lake is missing
https://dortania.github.io/OpenCore-Install-Guide/config.plist/

* Clearly explain the issue with the page
  
The excellent Dortania Guide is still at Comet Lake 10th-gen Intel, but people who are planning on buying new hardware have expressed reluctance on r/hackintosh to put together a system from two hardware generations ago. Alder Lake provides the biggest Intel performance jump since at least 2015. 

Currently there seems to be only limited support for Alder Lake systems  even though Dortania's OpenCore project has announced great progress with Z690:

[**Intel Z690 compatibility with macOS | Dortania**](https://dortania.github.io/hackintosh/updates/2022/01/09/alder-lake.html)  

There are very few Alder Lake OpenCore success stories on r/hackintosh. Here is one good example based on one of the guides from CaseySJ which I link to below:

[ASRock Z690 Pro RS with i5 12600k : hackintosh](https://www.reddit.com/r/hackintosh/comments/rps44y/asrock_z690_pro_rs_with_i5_12600k/)

The most detailed OpenCore based guides can be found on TM and the solutions found there could be used as a reasonable starting point for a beta draft version of a _Dortania OpenCore Guide for Alder Lake._

I have seen the following 'golden builds' recommended by r/hackintosh members:

[Asus Z690 ProArt Creator WiFi (Thunderbolt 4) + i7-12700K + AMD RX 6800 XT](https://www.tonymacx86.com/threads/asus-z690-proart-creator-wifi-thunderbolt-4-i7-12700k-amd-rx-6800-xt.318311/)

[Gigabyte Z690 Aero G + i5-12600K + AMD RX 6800 XT](https://www.tonymacx86.com/threads/gigabyte-z690-aero-g-i5-12600k-amd-rx-6800-xt.317179/)

Both are based on the Z690 technology discussion here:

[Z690 Chipset and Alder Lake CPUs](https://www.tonymacx86.com/threads/z690-chipset-and-alder-lake-cpus.316618/)

_The workflow is OpenCore based and could be easily adapted to the Dortania Guide workflow, IMHO._

Additionally, I have found two somewhat detailed OpenCore Alder Lake threads on hackintosh-forum.de in German, which use basically the same approach.

[Neuer Hackintosh mit ASUS PRIME Z690M-PLUS D4](https://www.hackintosh-forum.de/forum/thread/55711-neuer-hackintosh-mit-asus-prime-z690m-plus-d4/?pageNo=1)

I have checked InsanelyMac, but was disappointed with the approach of the few guides I found. I have not checked the other Dortania recommended forums, as Italian, Chinese and Russian is out of  reach for me. There is probably a lot more Alder Lake Hackintosh development work going on behind the scenes. 

* Include possible fixes

**I propose a beta draft version of a Dortania Alder Lake OpenCore Guide, which would spur more activity and development.** 

* Depending on the complexity of the fix, sources may be required

Sources, would be the links above especially the threads by CaseySJ.

**Comment by alyxferrari** [comment]
I have a local branch of the guide I'm working on for Alder Lake support.
@vit9696 Could you post a reference config for Z690 as you did for Z590 in #221? That would be very helpful. Thank you.

**Comment by vit9696** [comment]
@vandroiy2013 could you upload what we have now cleaned please?

**Comment by vandroiy2013** [comment]
@vit9696 Nothing new since this post https://dortania.github.io/hackintosh/updates/2022/01/09/alder-lake.html
CpuTopologyRebuild is still work in progress. Need more tests.

**Comment by vit9696** [comment]
I mean the plist file itself.

**Comment by vandroiy2013** [comment]
Here you go. 
[EFI-Z690-PUB.zip](https://github.com/dortania/bugtracker/files/8035267/EFI-Z690-PUB.zip)

**Comment by alyxferrari** [comment]
Thanks, I appreciate it.

**Comment by alyxferrari** [comment]
Is CpuTopologyRebuild required for anything? Also, SSDT-USBW is a companion to USBWakeFixup.kext, correct? I think these are my only questions.

**Comment by vit9696** [comment]
Hard to say, the research is ongoing. All in all there may be some minor power consumption increase at the cost of better single thread performance in very narrow scenarios. Your best is probably to say it exists and link to the repo, mentioning that it may positively or negatively affect one's performance, and that the experimentation is welcome.

**Comment by alyxferrari** [comment]
Yeah I saw some stuff on the tonymac forums about it right after I sent that which was why I deleted it. Thank you for the info, I'll mention that and the `-ctrsmt` boot arg, which I assume has something to do with that kext and hyperthreading?

**Comment by alyxferrari** [comment]
Nevermind, I found info on `-ctrsmt` in the CTR repo.

**Comment by chriswayg** [comment]
> Here you go. [EFI-Z690-PUB.zip](https://github.com/dortania/bugtracker/files/8035267/EFI-Z690-PUB.zip) OC 0.7.8

Which ASUS motherboard, CPU and GPU are you currently using this with?

**Comment by vandroiy2013** [comment]
ROG Strix Z690-F Gaming WIFI, i5-12600K and RX 6900 XT Reference.

**Comment by MiddlemanTM** [comment]
If Alder Lake is going to be added, I presume a similar guide for Rocket Lake should also be in order since that is missing too?

**Comment by alyxferrari** [comment]
@MiddlemanTM I've done that already, it's pending review. dortania/OpenCore-Install-Guide#343

**Comment by MiddlemanTM** [comment]
I thought to also mention here (for those who don't know) that conversion of a system to run Alder Lake requires the addition of modded SSDT-CPUR and SSDT-PLUG files because the naming convention has changed in the BIOS from 'Processor' to 'Processor Device' ie. CPU to CP01. You need a 'CPU device wrapper' in the ACPI in order to successfully boot macOS.

I have just discovered this new naming convention is also present in some of the newer Ryzen systems like the Lenovo Legion 5 with Ryzen 5800H & Radeon 6600M (which has successfully booted into Big Sur according to one poster I've spoken to).

**Comment by alyxferrari** [comment]
Yep, my Rocket and Alder Lake PRs have `SSDT-PLUG-ALT`. Thanks for the reminder though 🙂

**Comment by alyxferrari** [comment]
The Dortania team is probably just busy. AFAIK there's not much left, if anything, to do though.

**Comment by jacox02** [comment]
I may be blind, but,  I think Alder Lake and Alder Lake Laptop guide still missing on the guide...

**Comment by tarbaII** [comment]
1Revenger1 and I are working on getting Alder Lake added. Alder Lake laptops are not possible.

**Comment by jacox02** [comment]
> 1Revenger1 and I are working on getting Alder Lake added. Alder Lake laptops are not possible.

Im sad of hearing that, I just bought a brand new Asus Laptop with this new processor thinking on hackingtosh it.

**Comment by kabirbg** [comment]
I don't mean to necrobump, but is there any update on this? It's been 2 years since Alder Lake came out, and @tarbaII's guide looks practically ready for use. Is there a reason it isn't on the official site yet (which also doesn't seem to have been updated for months)—I don't mean to be rude, but just wondering!

**Comment by alyxandraio** [comment]
There are still unresolved reviews on most of the PRs, last I checked. I've been pretty busy this past year so it's a low priority for me at the moment. Hopefully by the time February rolls around I'll be able to get this fixed up. In the meantime, contributions are always welcome.

**Comment by kabirbg** [comment]
Thank you, and totally understandable!! I tried to see and it seems like the two reviews have all been resolved, am I just looking in the wrong place? I'd love to help contribute wherever I can, but I'm having trouble seeing what exactly is left to be done so I would appreciate it if you could point me in the right direction!

**Comment by santaclose666** [comment]
Last month, I bought a 12400f CPU and was looking forward to installing hackintosh on it but there still seems to be no easy-to-understand instructions for Alder Lake for newbies like me. Hopefully there will be many new updates this year. Thank you to those who contributed and wishing you guys lots of health.
